### DISCUSSION TITLE
How to adjust temperature readings from new probe

### DISCUSSION THREAD

**Discussion by bagelsbrews** [artisan-roaster-scope/artisan#1119] [General]
Diedrich IR-7.  It appears that the LCD displaying the original bean probe on this machine is more accurate than the new 3mm probe I just installed only 25mm away.  Here's a few logged temperature differences - 

**Note**                          **original probe**       **new probe**
first drop                            400                              420
                                           270                              348
                                           325                              388
                                           380                              427
 first crack                           388                             435

Is there a way to program an equation into Artisan so that it adjusts the temperature to the original probe temperature.   My thinking is that if nothing else the FC temperature shows the issue.

Thank you,

Mike

**Comment by poundy** [comment]
there's about two hundred posts (#exaggeration) on how to adjust this, IF you want to. Here's one https://github.com/artisan-roaster-scope/artisan/discussions/1082 that points to the doco on how you can symbolic temp formulae if you decide to adjust (which you don't really need to..... you just need to adjust your thinking)

**Reply by Dgocof** [reply]
It appears that your new probe is sensing more air temp than bean temp. The difference between the temperatures is not the same from drop to first crack so a formula probably won't work for your situation. If the probes are different thickness then the temperature rise will not be the same from TP to end of roast. I'm dealing with the same issue.

**Reply by MAKOMO** [reply]
How can you know which of the two probes is more accurate? I both are just an approximation of the bean temperature (or rather a bean temperature / air temperature / drum temperature mix at a given,  maybe not even constant, ratio.

A simple offset will not help you in this case, but it seems that a linear approximation should get your Artisan readings closer to the ones displayed by the machine. Luckily, Artisan allows you to use formula of any complexity. 

My take: add the formula

```x*1.37 -206.91```

as symbolic formula under "BT Y(x)" in the Symb ET/BT tab of the Device Assignment dialog (menu `Config >> Device`) as in the following screen shot.

<img width="572" alt="Screenshot 2023-03-04 at 13 40 29" src="https://user-images.githubusercontent.com/13909011/222902023-537a415f-40a7-4f94-85de-b814639ffe90.png">


This formula above results from a linear regression calculated by Excel (or Apple Numbers or OpenOffice,...) as follows.

<img width="458" alt="Screenshot 2023-03-04 at 13 39 12" src="https://user-images.githubusercontent.com/13909011/222902060-b152069e-3c3c-4705-8fbb-df61719005d2.png">

The linear approximation in a scatter chart taking all of the readings looks like this.

![all_readings](https://user-images.githubusercontent.com/13909011/222902146-02b2eb36-2243-4cc5-939a-4a836b80cf29.png)

Not too bad, but obviously that first reading at CHARGE is off. Most likely as this situation reading just air temperature is different than during the roast where a certain mix is measured. Leaving off this first reading allows for an almost perfect linear mapping (R^2=0,9995).

![all_but_first_readings](https://user-images.githubusercontent.com/13909011/222902205-ce8a568f-0da8-4e93-a512-a3ef06f4f49c.png)

The 4 hidden formulas to calculate the offset and slope of the mapping are as follows:

- taking all readings

Offset: `INTERCEPT(C3:C7;B3:B7)`
Slope: `SLOPE(C3:C7;B3:B7)`

- taking all but the first

Offset: `INTERCEPT(C4:C7;B4:B7)`
Slope: `SLOPE(C4:C7;B4:B7)`

I converted the Numbers file to OpenOffice format and attached it below. You just need to remove the `.txt` filename extension.

[Uploading tempMapping.ods.txt…]()

More readings could improve this mapping. Maybe calculating a quadratic mapping would be even better.

But I agree with [poundy](https://github.com/poundy) above: you just need to adjust your thinking
